### ISSUE TITLE
Python SDK has limitation on concurrent queries?

### ISSUE THREAD

**Issue by vrlb** [Azure/azure-kusto-python#220]
#### Code Sample, a copy-pastable example if possible

```python

import os, uuid, timeit, functools
import numpy
from query import get_query
from azure.kusto.data.request import KustoClient, KustoConnectionStringBuilder, ClientRequestProperties

db_name = os.environ.get("DATABASE_NAME")
test_id = os.environ.get("TEST_ID", str(uuid.uuid4()))
query_consistency = os.environ.get("QUERY_CONSISTENCY", "weakconsistency")

request_properties = ClientRequestProperties()
request_properties.set_option("queryconsistency", query_consistency)

print("Test run for '{}' started.".format(test_id))

def execute_query(raw_query):
    query = "{} //TEST_ID={}".format(raw_query, test_id)
    response = kusto_client.execute(db_name, query, request_properties)

queries_total = int(os.environ.get("QUERIES_TOTAL", -1))
queries_executed = 0

while queries_executed < queries_total or queries_total < 0:
    raw_query = get_query()
    print("\nTest '{}' executing #{}:\n{}\n".format(test_id, queries_executed, raw_query))

    t = timeit.Timer(functools.partial(execute_query, raw_query))
    query_time = t.timeit(number=1)

    print("Query took: {:.2f} seconds".format(query_time))
    queries_executed += 1

print("Test run for '{}' ended.".format(test_id))
```
#### Problem description

I am running this script to test the query time by using 100 pods in AKS running at the same time. I didn't set the queries total, so it is supposed to keep running and querying. During the test, AKS pods show a patten of pause when they are running for a few minutes, and restart again. Per check, there is no capacity issue in both AKS and ADX compute. I am investigating other reasons of this behaving and suspect if this is due to the API call in query/v2 when executing kustoClient.execute(). Also, the time needed to run queries is higher when the concurrent queries dropped to 1. 

![image](https://user-images.githubusercontent.com/49311980/78867605-efba1980-7a73-11ea-842f-7b8a58dd7394.png)
![image](https://user-images.githubusercontent.com/49311980/78869310-f39b6b00-7a76-11ea-9aae-3103acef4f5b.png)

Other factors are eliminated including:
- I used a 5 nodes AKS cluster, all nodes have no capacity issue - CPU < 5%, memory <20%
- ADX cluster is healthy -  CPU <30%, 4 instances, D14v2 SKU 16 vCores
- Average queries time is stable around 35ms. Dataset size is just about 49.96 MB. 
- All the pods are using same variables, same AAD and same code for building connection to ADX.
`KustoConnectionStringBuilder.with_aad_application_key_authentication(
    cluster, client_id, client_secret, tenant_id)`

**Questions:**

1. Is there any limitation about parallel queries when I use kustoClient.execute()? 
2. For each call of ClientRequestProperties(), does it has limitation in terms of number of call per seconds/minutes/connection?
3. Are there other factors that possibly affect the time to run KustoClient in azure.kusto.data.request ?
4. Any suggestion on alternative queries testing for concurrent queries? 

--
#### If query related, does it happen on other platforms (Kusto Web UI, Kusto Explorer)? 
same pattern occurred in Kusto web UI when i check the .show queries and it shows the same pattern of sudden decline and sudden rise.

**Comment by vladikbr** [comment]
Query concurrency is a limit that a cluster imposes on a number of queries running at the same time. The default value of the query concurrency limit depends on the SKU cluster it is running on, and is calculated as: Cores-Per-Node x 10. For example, for a cluster that's set-up on D14v2 SKU, where each machine has 16 vCores, the default Query Concurrency limit will be 16 cores x10 = 160. The default value can be changed by creating a support ticket. In the future, this control will also be exposed via a control command.

**Comment by vrlb** [comment]
@vladikbr Yes I have checked this document, that's why I have provisioned a cluster of 16 cores, the Query Concurrency limit should be 160. But from the graph and metrics, it seems like it can't even handle 100. The concurrent queries shows only a maximum of 14. I have also provided that it seems unlikely to be issues around compute resources , with CPU <30%.

-> is there any other constraints for parallel queries with same AAD if I execute query through this SDK? Is there any limitation about parallel queries when I use kustoClient.execute()?

thanks!
